controllers/managedocs_controller.go

```diff
@@ -2045,3 +2096,31 @@ func (r *ManagedOCSReconciler) setDeviceSetCount(deviceSet *ocsv1.StorageDeviceS
 		deviceSet.Count = currDeviceSetCount
 	}
 }
+
+func (r *ManagedOCSReconciler) reconcileODFSubscription() error {
+	subList := &opv1a1.SubscriptionList{}
+	if err := r.list(subList); err != nil {
+		return fmt.Errorf("Unable to list subscriptions %v", err)
+	}
+	odfSub := &opv1a1.Subscription{}
+	for index := range subList.Items {
+		item := &subList.Items[index]
+		if item.Spec.Package == odfOCMBundleName {
+			odfSub = item
+			break
+		}
+	}
+	if odfSub == nil {
```

Comment:
There is a bug here:

This will never be nil because you are initializing it with `odfSub := &opv1a1.Subscription{}` which means it has a default value.

You should initialize it as 
```
var odfSub *opv1a1.Subscription = nil
```
Instead

Reply:
Updated